src/main/scala/nl/knaw/dans/easy/fedora2vault/EasyFedora2vaultApp.scala

```diff
@@ -116,7 +125,8 @@ class EasyFedora2vaultApp(configuration: Configuration) extends DebugEnhancedLog
         .map(compareManifest(bag))
         .getOrElse(Success(())) // TODO check with sha's from fedora
       doi = emd.getEmdIdentifier.getDansManagedDoi
-    } yield FeedBack(datasetId, doi, depositor, TransformationType.SIMPLE, uuid, "???").toString
+      msg <- CsvRecord(datasetId, doi, depositor, SIMPLE, UUID.fromString(bagDir.name), "OK").print
+    } yield msg
```

Comment:
How about you yield the `CsvRecord` here, return it to `Command` and do the printing there. That makes your code more testable and removes the dependency upon the `CSVPrinter`.
This also opens up the way to extend `CsvRecord` (perhaps renamed back to `Feedback`) with other methods that output in a different format.

Reply:
I've been looking into that before. Ran into trouble with failing fast (`collectFirst`) on fatal errors in `simpleTransForms`.

Reply:
Let me postpone that together with improved error handling to a new pull request. With this one merged I can create a pull request for files.xml, then @lindareijnhoudt can try https://drivenbydata.atlassian.net/browse/EASY-2709

Reply:
👍